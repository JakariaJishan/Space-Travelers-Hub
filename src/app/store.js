import { configureStore } from '@reduxjs/toolkit';
import missionsSlice from './missions/missionsSlice';
import rocketsSlice from './missions/rocketsSlice';

export default store = configureStore({
  reducer: missionsSlice, rocketsSlice,
});
