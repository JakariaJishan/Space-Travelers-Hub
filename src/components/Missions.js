import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { joinMissions, leaveMissions } from '../app/missions/missionsSlice';
import RenderMissions from './RenderMissions';
import '../style/missions.css';

function Missions() {
  const missions = useSelector((state) => state.missions.missions);
  const dispatch = useDispatch();

  const handleJoin = (id) => {
    dispatch(joinMissions(id));
  };
  const handleLeave = (id) => {
    dispatch(leaveMissions(id));
  };

  return (
    <div className="otro">
      <div className="mission_container">
        <h2 className="table"> Mission</h2>
        <h2 className="table"> Description</h2>
        <h2 className="table"> Status</h2>
        <h2 className="table"> </h2>
      </div>
      {missions.map((mission) => (
        <RenderMissions
          key={mission.mission_id}
          mission={mission}
          handleJoin={handleJoin}
          handleLeave={handleLeave}
        />
      ))}
    </div>
  );
}

export default Missions;
