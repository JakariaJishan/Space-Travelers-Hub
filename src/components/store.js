import { configureStore } from '@reduxjs/toolkit';
import missionsSlice from './missions/missionsSlice';
import rocketsSlice from './missions/rocketsSlice';

const store = configureStore({
  reducer: missionsSlice,
  rocketsSlice,
});

export default store;
