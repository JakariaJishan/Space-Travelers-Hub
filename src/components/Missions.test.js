import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../app/store';
import Missions from './Missions';

const mission = {
  mission_id: 'qrcae42',
};

describe('testing Missions', () => {
  it('test component renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions
            key={mission.mission_id}
            mission={mission}
          />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
