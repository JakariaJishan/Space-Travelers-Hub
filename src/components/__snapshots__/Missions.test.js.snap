// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Missions test component renders correctly 1`] = `
<div
  className="otro"
>
  <div
    className="mission_container"
  >
    <h2
      className="table"
    >
       Mission
    </h2>
    <h2
      className="table"
    >
       Description
    </h2>
    <h2
      className="table"
    >
       Status
    </h2>
    <h2
      className="table"
    >
       
    </h2>
  </div>
</div>
`;
